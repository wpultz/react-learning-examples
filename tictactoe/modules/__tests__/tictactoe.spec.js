import { PLAYER_X, PLAYER_O, EMPTY } from '../../constants'
import reducer, { makeMove, reset } from '../tictactoe'

it('should have default state', () => {
  const state = reducer(undefined, { type: 'INIT' })

  expect(state).toMatchSnapshot()
})

it('should react to a move by X', () => {
  let state = reducer(undefined, { type: 'INIT' })

  expect(state).toMatchSnapshot()

  state = reducer(state, makeMove(PLAYER_X, 0))

  expect(state).toMatchSnapshot()
})

it('should react to a move, then be reset', () => {
  let state = reducer(undefined, { type: 'INIT' })

  expect(state).toMatchSnapshot()

  state = reducer(state, makeMove(PLAYER_O, 5))

  expect(state).toMatchSnapshot()

  state = reducer(state, reset())

  expect(state).toMatchSnapshot()
})