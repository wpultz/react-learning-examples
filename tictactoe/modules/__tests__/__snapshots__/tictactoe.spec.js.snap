// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have default state 1`] = `
Object {
  "currentPlayer": "X",
  "positions": Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
}
`;

exports[`should react to a move by X 1`] = `
Object {
  "currentPlayer": "X",
  "positions": Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
}
`;

exports[`should react to a move by X 2`] = `
Object {
  "currentPlayer": "O",
  "positions": Array [
    "X",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
}
`;

exports[`should react to a move, then be reset 1`] = `
Object {
  "currentPlayer": "X",
  "positions": Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
}
`;

exports[`should react to a move, then be reset 2`] = `
Object {
  "currentPlayer": "X",
  "positions": Array [
    "",
    "",
    "",
    "",
    "",
    "O",
    "",
    "",
    "",
  ],
}
`;

exports[`should react to a move, then be reset 3`] = `
Object {
  "currentPlayer": "X",
  "positions": Array [
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
    "",
  ],
}
`;
